test/omnitrade_client.dart

```diff
@@ -0,0 +1,303 @@
+        expect(subject, expectedResult);
+      });
+    });
+  });
+
+  group('fetchTrades', () {
+    group('when market is invalid', () {
+      test('it returns empty list', () async {
+        final expectedResult = [];
+
+        final client = MockClient((request) async =>
+          Response(json.encode(expectedResult), 200)
+        );
+
+        final omniClient = OmniTradeClient(httpClient: client);
+        final subject = (await omniClient.fetchTrades('anything')).body;
+        expect(subject, expectedResult);
+      });
+    });
+
+    group('when market valid', () {
+      test('it returns list with recent trades', () async {
+        final expectedResult = [
+          {
+            "tid": 1618491,
+            "price": "12700.0",
+            "amount": "0.00315992",
+            "funds": "40.130984",
+            "market": "btcbrl",
+            "created_at": "2019-02-06T14:42:51-02:00",
+            "date": 1549471371,
+            "side": null
+          },
+        ];
+
+        final client = MockClient((request) async =>
+          Response(json.encode(expectedResult), 200)
+        );
+
+        final omniClient = OmniTradeClient(httpClient: client);
+        final subject = (await omniClient.fetchTrades('btcbrl')).body;
+        expect(subject, expectedResult);
+      });
+    });
+  });
+
+  group('fetchK', () {
+    group('when market is invalid', () {
+      test('it returns http status 404', () async {
+        final expectedResult = {
+          "error": {
+            "code": 1002,
+            "message": "Couldn't find Market with 'id'=anything"
+          }
+        };
+
+        final client = MockClient((request) async =>
+            Response(json.encode(expectedResult), 404)
+        );
+
+        final omniClient = OmniTradeClient(httpClient: client);
+        expect(omniClient.fetchK('anything'), throws);
+      });
+    });
+
+    group('when market valid', () {
+      test('it returns list with OHLC(k line) of specific market', () async {
+        final expectedResult = [
+          [
+            1549469940,
+            12700,
+            12700,
+            12700,
+            12700,
+            0
+          ],
+        ];
+
+        final client = MockClient((request) async =>
+          Response(json.encode(expectedResult), 200)
+        );
+
+        final omniClient = OmniTradeClient(httpClient: client);
+        final subject = (await omniClient.fetchK('btcbrl')).body;
+        expect(subject, expectedResult);
+      });
+    });
+  });
+
+  group('fetchKWithPendingTrades', () {
+    group('when market is invalid', () {
+      test('it returns http status 404', () async {
+        final expectedResult = {
+          "error": {
+            "code": 1002,
+            "message": "Couldn't find Market with 'id'=anything"
+          }
+        };
+
+        final client = MockClient((request) async =>
+          Response(json.encode(expectedResult), 404)
```

Comment:
You should test if it's using the correct API path here. The way it is right now, if the `fetchKWithPendingTrades` function uses the wrong path the test wouldn't fail, right?